File: Laborator/CSharp/AgentieTurism/Tests/TicketRepositoryTest.cs
using System;
using System.Collections.Generic;
using System.IO;
using NUnit.Framework;
using AgentieTurism.Models;
using AgentieTurism.Repository;

namespace AgentieTurism.Tests
{
    [TestFixture]
    public class TicketRepositoryTest
    {
        private static ITicketRepository ticketRepo;
        private static IFlightRepository flightRepo;
        private static Flight flight;
        private static Ticket ticket;

        [OneTimeSetUp]
        public void Setup()
        {
            //var props = new Dictionary<string, string>();
            //foreach (var line in File.ReadLines("bd.config"))
            //{
            //    var parts = line.Split('=');
            //    if (parts.Length == 2)
            //        props[parts[0].Trim()] = parts[1].Trim();
            //}

            //flightRepo = new FlightDbRepository(props);
            //ticketRepo = new TicketDbRepository(props);
            flightRepo = new FlightDbRepository();
            ticketRepo = new TicketDbRepository();

            flight = new Flight("New York", new DateTime(2025, 7, 20, 18, 45, 0), "JFK", 250);
            flightRepo.Add(flight);
        }

        [Test, Order(1)]
        public void TestAddTicket()
        {
            ticket = new Ticket(flight, "John Doe", "Alice, Bob", "123 Main St", 2);
            ticketRepo.Add(ticket);
            Assert.That(ticket.Id, Is.Not.Null);
        }

        [Test, Order(2)]
        public void TestFindTicketById()
        {
            var foundTicket = ticketRepo.FindOne(ticket.Id);
            Assert.That(foundTicket, Is.Not.Null);
            Assert.That(foundTicket.ClientName, Is.EqualTo("John Doe"));
        }

        [Test, Order(3)]
        public void TestFindByFlightId()
        {
            var tickets = ticketRepo.FindByFlightId(flight.Id);
            Assert.That(tickets, Is.Not.Empty);
        }

        [Test, Order(4)]
        public void TestDeleteTicket()
        {
            ticketRepo.Delete(ticket.Id);
            Assert.That(ticketRepo.FindOne(ticket.Id), Is.Null);
        }
    }
}

File: Laborator/Java/AgentieTurism/src/test/java/TicketRepositoryTest.java
import org.junit.jupiter.api.*;
import ro.mpp2024.domain.Flight;
import ro.mpp2024.domain.Ticket;
import ro.mpp2024.repository.FlightDBRepository;
import ro.mpp2024.repository.FlightRepository;
import ro.mpp2024.repository.TicketDBRepository;
import ro.mpp2024.repository.TicketRepository;

import java.io.FileReader;
import java.io.IOException;
import java.time.LocalDateTime;
import java.util.List;
import java.util.Properties;

import static org.junit.jupiter.api.Assertions.*;

@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
public class TicketRepositoryTest {
    private static TicketRepository ticketRepo;
    private static FlightRepository flightRepo;
    private static Flight flight;
    private static Ticket ticket;

    @BeforeAll
    static void setup() {
        Properties props = new Properties();
        try {
            props.load(new FileReader("bd.config"));
        } catch (IOException e) {
            fail("Cannot find bd.config: " + e);
        }
        flightRepo = new FlightDBRepository(props);
        ticketRepo = new TicketDBRepository(props);

        flight = new Flight("New York", LocalDateTime.of(2025, 7, 20, 18, 45), "JFK", 250);
        flightRepo.add(flight);
    }

    @Test
    @Order(1)
    void testAddTicket() {
        ticket = new Ticket(flight, "John Doe", "Alice, Bob", "123 Main St", 2);
        ticketRepo.add(ticket);
        assertNotNull(ticket.getId());
    }

    @Test
    @Order(2)
    void testFindTicketById() {
        Ticket foundTicket = ticketRepo.findOne(ticket.getId());
        assertNotNull(foundTicket);
        assertEquals("John Doe", foundTicket.getClientName());
    }

    @Test
    @Order(3)
    void testFindByFlightId() {
        List<Ticket> tickets = ticketRepo.findByFlightId(flight.getId());
        assertFalse(tickets.isEmpty());
    }

    @Test
    @Order(4)
    void testDeleteTicket() {
        ticketRepo.delete(ticket.getId());
        assertNull(ticketRepo.findOne(ticket.getId()));
    }
}